### PR TITLE
Change default hive metastore refresh interval

### PR DESCRIPTION
**Pull request by s2lomon**
With previous configuration, whenever someone was setting up
a ttl, refresh-interval default was used as 0, which is an invalid
configuration option for cache, which in turn has caused PrestoServer
startup failure. Changing it to Duration(Integer.MAX_VALUE, DAYS)
makes it disabled by default as well, but allows configuration
with ttl only, as refresh interval is not applied when it's greater than ttl.

### CODE REVIEW

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java [hunk 1]
```diff
@@ -263,6 +274,20 @@ public void testNoCacheExceptions()
         assertEquals(mockClient.getAccessCount(), 2);
     }
 
+    @Test
+    public void testServiceCreationWithOnlyTTLEnabled()
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: testServiceCreationWithOnlyTTLEnabled -> testCachingHiveMetastoreCreationWithTtlOnly?

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java [hunk 2]
```diff
@@ -69,6 +69,17 @@ public void setUp()
         stats = thriftHiveMetastore.getStats();
     }
 
+    private ListeningExecutorService createTestThreadPool()
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: undo, it is used only in single place

Comment (s2lomon) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: in 2 places

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java (hunk 1 above)
Comment (s2lomon) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: sounds ok

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java (hunk 2 above)
Comment (kokosing) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: Right, I forgot to comment the other usage ;)

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java [hunk 3]
```diff
@@ -263,6 +274,20 @@ public void testNoCacheExceptions()
...
 
+    @Test
+    public void testServiceCreationWithOnlyTTLEnabled()
+    {
+        CachingHiveMetastoreConfig config = new CachingHiveMetastoreConfig();
+        config.setMetastoreCacheTtl(new Duration(10, TimeUnit.MILLISECONDS));
+
+        CachingHiveMetastore metastore = new CachingHiveMetastore(
+                new BridgingHiveMetastore(createThriftHiveMetastore()),
+                createTestThreadPool(),
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: Please use direct executor here.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastoreConfig.java [hunk 4]
```diff
@@ -25,7 +25,7 @@
 public class CachingHiveMetastoreConfig
 {
     private Duration metastoreCacheTtl = new Duration(0, TimeUnit.SECONDS);
-    private Duration metastoreRefreshInterval = new Duration(0, TimeUnit.SECONDS);
+    private Duration metastoreRefreshInterval = new Duration(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
```
[the remark this answers is not in the document]
Comment (s2lomon) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastoreConfig.java]: hm ok, it can be DAYS as well

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java (hunk 3 above)
Comment (s2lomon) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: I would wrap it all in separate methods to increase the fluency of reading these tests, but I know how much you don't like that ;) 
`CachingHiveMetastore createMetastore(...)`

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java [hunk 5]
```diff
@@ -263,6 +270,25 @@ public void testNoCacheExceptions()
...
+        CachingHiveMetastore metastore = createMetastore(config);
+
+        assertThat(metastore).isNotNull();
+    }
+
+    private CachingHiveMetastore createMetastore(CachingHiveMetastoreConfig config)
+    {
+        return new CachingHiveMetastore(
+                new BridgingHiveMetastore(createThriftHiveMetastore()),
+                MoreExecutors.directExecutor(),
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: static import

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java [hunk 6]
```diff
@@ -263,6 +270,25 @@ public void testNoCacheExceptions()
...
+    {
+        CachingHiveMetastoreConfig config = new CachingHiveMetastoreConfig();
+        config.setMetastoreCacheTtl(new Duration(10, TimeUnit.MILLISECONDS));
+
+        CachingHiveMetastore metastore = createMetastore(config);
+
+        assertThat(metastore).isNotNull();
+    }
+
+    private CachingHiveMetastore createMetastore(CachingHiveMetastoreConfig config)
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: I would inline this method. It creates metastore with direct executor and so it makes it usable only from one context (the test above).

Alternatively, please rename createMetastore -> createMetastoreWithDirectExecutor

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java [hunk 7]
```diff
@@ -57,9 +59,8 @@
     public void setUp()
     {
         mockClient = new MockThriftMetastoreClient();
-        MetastoreLocator metastoreLocator = new MockMetastoreLocator(mockClient);
+        ThriftHiveMetastore thriftHiveMetastore = createThriftHiveMetastore();
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: it looks like changes here are not needed by the actual change, am I missing something? Either extract them as separate commit (or even PR because I am not sure if they are worth doing) or simply drop.

Comment (s2lomon) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: I'm using the createThriftHiveMetastore method when constructing the CachingHiveMetastore, as I don't want to introduce any stubs and using actual implementation seems fine here.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java (hunk 6 above)
Comment (s2lomon) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/cache/TestCachingHiveMetastore.java]: I've renamed the method as it sounds ok for the test purpose

**Comment by electrum** [comment]
The tests are failing due to `TestCachingHiveMetastoreConfig` needing to be updated. Otherwise looks good.

**Comment by kokosing** [comment]
Merged, thanks!
